lib/core-integration/src/Test/Integration/Scenario/API/Network.hs

```diff
@@ -162,77 +162,92 @@ spec = do
                     Link.getNetworkInfo headers Empty
                 verify r expectations
 
-    describe "NETWORK - Cannot query blockchain parameters with \
-             \invalid arguments" $ do
-        let epochNoWrong =
-                T.pack $show $ (+10) $ fromIntegral @Word31 @Int maxBound
-        let matrix = ["earliest", "invalid", epochNoWrong, "-1"]
+    describe "NETWORK_PARAMS_01 - Valid epoch values" $ do
+        let matrix = ["latest", "0"]
+        forM_ matrix $ \epochNo -> it ("Epoch: " <> show epochNo) $ \ctx -> do
+            verifyEpochNumOK ctx epochNo
+
+        it "Current epoch" $ \ctx -> do
+            r <- request @ApiNetworkInformation ctx
+                Link.getNetworkInfo Default Empty
+            let currentEpochNo =
+                    getFromResponse
+                        (#nodeTip . #epochNumber . #getApiT . #unEpochNo) r
+            let epochNo = T.pack $ show currentEpochNo
+            verifyEpochNumOK ctx epochNo
+
+        it "Previous epoch" $ \ctx -> do
+            r <- request @ApiNetworkInformation ctx
+                Link.getNetworkInfo Default Empty
+            let currentEpochNo =
+                    getFromResponse
+                        (#nodeTip . #epochNumber . #getApiT . #unEpochNo) r
+            -- test previous epoch unless the current is epoch 0
+            -- otherwise test hits maxBound of the epochNo and fails
+            epochNo <- if (currentEpochNo > 0) then do
+                return $ T.pack $ show (currentEpochNo - 1)
+            else do
+                return $ T.pack $ show currentEpochNo
+
+            verifyEpochNumOK ctx epochNo
+
+    describe "NETWORK_PARAMS_02 - Cannot query future epoch" $  do
+        it "Future epoch" $ \ctx -> do
+            r1 <- request @ApiNetworkInformation ctx
+                Link.getNetworkInfo Default Empty
+            let (ApiT (EpochNo currentEpochNo)) =
+                    getFromResponse (#nextEpoch . #epochNumber) r1
+
+            let futureEpochNo = T.pack $ show $ currentEpochNo + 10
+            let endpoint = ( "GET", "v2/network/parameters/"<>futureEpochNo )
+            r2 <- request @ApiNetworkParameters ctx endpoint Default Empty
+
+            expectResponseCode @IO HTTP.status404 r2
+            expectErrorMessage (errMsg404NoEpochNo (T.unpack futureEpochNo)) r2
```

Comment:
this one also can be reused with `Epoch max value`

Reply:
extracted `verifyEpochNumWrong` in 96bf986e189c3c361937ce5eedc91a01d44daff6.
I think nothing else left/makes really sense to extract :)